### PR TITLE
Java 11 readiness: build both on JDK8 and JDK11

### PR DESCRIPTION
**Pull request by batmat**
Work in progress to make sure this plugin is constantly checked both building with JDK8 and JDK11.

This goes with the Jenkins Java 11 General Availability announcement made back in March, and we're making a pass to check plugins are looking good on a JDK11.

([to understand the Jenkinsfile content](https://wiki.jenkins.io/display/JENKINS/Java+11+Developer+Guidelines#Java11DeveloperGuidelines-MakesureyourpluginistestedinContinuousIntegrationonJava8andJava11atthesametime))

@jenkinsci/java11-support

### CODE REVIEW

**Comment by kuisathaverat** [comment]
I was waiting for this one ;) In my case the defaults are not good for me, I support only the latest 2 LTS and the latest Weekly, I've to change this file a lot of times. The default only tests 2 LTS, If I merge this one I lose the weekly test.

**Comment by batmat** [comment]
No worries. My bad, I filed a few similar PRs and didn't check enough. Will fix this PR by actually merging both recommended configuration + the more recent weekly you want to test. :+1:

File: Jenkinsfile [hunk 1]
```diff
@@ -1,8 +1,7 @@
...
-  [ platform: "windows", jdk: "8", jenkins: "2.164.1", javaLevel: "8" ],
-  [ platform: "linux", jdk: "11", jenkins: "2.168", javaLevel: "8" ],
-  [ platform: "windows", jdk: "11", jenkins: "2.168", javaLevel: "8" ]
-])
\ No newline at end of file
+def buildConfiguration = buildPlugin.recommendedConfigurations()
+
+// Also build on recent weekly
+buildConfiguration << [ platform: "linux",   jdk: "11", jenkins: "2.168", javaLevel: "8" ]
+buildConfiguration << [ platform: "windows", jdk: "11", jenkins: "2.168", javaLevel: "8" ]
```
Comment (batmat) [inline Jenkinsfile]: @kuisathaverat I believe that should do the trick. Let's see what CI says :)

**Comment by kuisathaverat** [comment]
I resolved the conflict I trigger by merging a couple of PRs, if the CI pass It is good for me :)

**Pull request merged**